components/Header/Header.tsx

```diff
@@ -9,13 +9,11 @@ import Tooltip from "../ui/Tooltip";
 
 const Header = () => {
   const { push } = useRouter();
-  const { address, connectWallet } = useConnectWallet();
+  const { authenticated, ready } = useConnectWallet();
 
   const handleClick = (link: string) => {
-    if (address) {
+    if (authenticated && ready) {
       push(link);
-    } else {
-      connectWallet();
```

Comment:
we still want to trigger `login` on header button click for logged out user.
- why are you removing this `else`?